OSC support

piegamesde opened the pull request:
- Added some simple microtonal support for the synthesizer
- Added some OSC commands and dispatch
- This won't compile until the build system has been updated. I tried adding `rtosc` as dependency, but C build systems are weird.
    - Even after updating, it probably won't work (I can't test it if it doesn't compile)
    - `rtosc` needs to be added as optional dependency. If the library is present as well as Jack support, `ENABLE_OSC` may be set to enable the feature.
- Implements #148
- Currently, only some synthesizer features are implemented, and no GUI connection is made. I don't know how much is needed to allow for all features requested in #86.

piegamesde committed:
OSC support

nickdowell commented:
Thanks for your work on this PR!

Can you give a bit of context around how to use this functionality - for example what's your test set-up?

piegamesde commented:
There is no test setup. This code barely even compiles*. Someone else needs to update the build system to handle the new rtosc dependency correctly.

\* I couldn't get it to build using autoconf. I changed a lot of stuff in `CMakelists.txt` until it compiled on my system, but the linker crashes with an error. Having two build systems that both generate Makefiles is really confusing.

nickdowell reviewed (commented):
Ok, understood. I'll take a stab at updating the build system (automake) to detect the new rtosc dependency, but it might be a few days before I find the time to do so.

piegamesde commented on src/JackOutput.cpp:
```diff
@@ -135,11 +141,20 @@ static inline amsynth_midi_event_t amsynth_midi_event_make(jack_midi_event_t je)
+static inline amsynth_osc_event_t amsynth_osc_event_make(jack_midi_event_t je) {
+	amsynth_osc_event_t oe;
+	oe.length = je.size;
+	oe.buffer = (char*) malloc(je.size);
```
Can I just point to the memory instead? This means all event data will be invalidated on the next processing frame. I don't know how much this is an issue.

nickdowell replied:
Yes that will work fine as long as the events are processed immediately (which they appear to be)

piegamesde commented:
Btw, the gitignore needs some love as well. Just building the project creates ~150 new untracked files on my system (mostly cmake stuff and binaries).

piegamesde committed:
Revert some changes

nickdowell committed:
Add rtosc to build system

nickdowell committed:
Add rtosc to Makefile.am

piegamesde committed:
Remove automake build system

piegamesde committed:
Make it build on my machine

piegamesde committed:
Update .gitignore

- Ignored VS Code projects
- Ignored Eclipse projects
- Ignore build output folder
- Remove autoconf ignores
- Remove binary ignores. If you see a binary that's not in `./output` or a similar folder, you're doing something wrong.

piegamesde committed:
Make it work
